packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -0,0 +1,759 @@
+  /// true.
+  ///
+  /// The default value is 256.
+  final double extendedWidth;
+
+  /// Returns the animation that controls the [NavigationRail.extended] state.
+  ///
+  /// This can be used to synchronize animations in the [leading] or [trailing]
+  /// widget, such as an animated menu or a [FloatingActionButton] animation.
+  static Animation<double> extendedAnimation(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType<_ExtendedNavigationRailAnimation>().animation;
+  }
+
+  @override
+  _NavigationRailState createState() => _NavigationRailState();
+}
+
+class _NavigationRailState extends State<NavigationRail> with TickerProviderStateMixin {
+  List<AnimationController> _destinationControllers = <AnimationController>[];
+  List<Animation<double>> _destinationAnimations;
+  AnimationController _extendedController;
+  Animation<double> _extendedAnimation;
+
+  @override
+  void initState() {
+    super.initState();
+    _initControllers();
+  }
+
+  @override
+  void dispose() {
+    _disposeControllers();
+    super.dispose();
+  }
+
+  @override
+  void didUpdateWidget(NavigationRail oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    if (widget.extended != oldWidget.extended) {
+      if (widget.extended) {
+        _extendedController.forward();
+      } else {
+        _extendedController.reverse();
+      }
+    }
+
+    // No animated segue if the length of the items list changes.
+    if (widget.destinations.length != oldWidget.destinations.length) {
+      _resetState();
+      return;
+    }
+
+    if (widget.currentIndex != oldWidget.currentIndex) {
+      _destinationControllers[oldWidget.currentIndex].reverse();
+      _destinationControllers[widget.currentIndex].forward();
+      return;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final NavigationRailThemeData navigationRailTheme = NavigationRailTheme.of(context);
+    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+
+    final Color backgroundColor = widget.backgroundColor ?? navigationRailTheme.backgroundColor ?? theme.colorScheme.surface;
+    final double elevation = widget.elevation ?? navigationRailTheme.elevation ?? 0;
+    final Color baseSelectedColor = theme.colorScheme.primary;
+    final Color baseColor = theme.colorScheme.onSurface.withOpacity(0.64);
+    final IconThemeData unselectedIconTheme = const IconThemeData.fallback().copyWith(color: baseColor).merge(widget.unselectedIconTheme ?? navigationRailTheme.unselectedIconTheme);
+    final IconThemeData selectedIconTheme = const IconThemeData.fallback().copyWith(color: baseSelectedColor).merge(widget.selectedIconTheme ?? navigationRailTheme.selectedIconTheme);
+    final TextStyle unselectedLabelTextStyle = TextStyle(color: baseColor, fontSize: 14.0).merge(widget.unselectedLabelTextStyle ?? navigationRailTheme.unselectedLabelTextStyle);
```

Comment:
Should the defaults not pull from the 2018 text theme instead?

Reply:
What do you mean?

Reply:
Like rather than create a new `TextStyle`, shouldn't it use like `Theme.of(context).textTheme.body1` or whichever maps to the right value? I would think that the spec would have it map to one of the text theme values.

Reply:
Done, good idea